File: src/main/java/ru/ioleksiv/telegram/bot/core/controller/annotations/parser/unpacker/EditedMessageUnpackerParser.java
package ru.ioleksiv.telegram.bot.core.controller.annotations.parser.unpacker;

import org.springframework.stereotype.Component;
import ru.ioleksiv.telegram.bot.api.annotations.receiver.EditedMessageReceiver;
import ru.ioleksiv.telegram.bot.api.model.objects.std.Message;
import ru.ioleksiv.telegram.bot.core.controller.handler.unpack.UpdateUnpacker;

import java.util.Optional;

@Component
public class EditedMessageUnpackerParser extends UnpackerParser<EditedMessageReceiver, Message> {

    @Override
    public Class<EditedMessageReceiver> getAnnotationClass() {
        return EditedMessageReceiver.class;
    }

    @Override
    public UpdateUnpacker<Message> getUpdateUnpacker() {
        return new UpdateUnpacker<>(in -> Optional.ofNullable(in.getEditedMessage()), Message.class);
    }

}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/core/controller/annotations/parser/unpacker/EditedMessageUnpackerParser.kt
package io.github.oleksivio.telegram.bot.core.controller.annotations.parser.unpacker

import io.github.oleksivio.telegram.bot.api.annotations.receiver.EditedMessageReceiver
import io.github.oleksivio.telegram.bot.api.model.objects.std.Message
import io.github.oleksivio.telegram.bot.core.controller.handler.unpack.UpdateUnpacker
import org.springframework.stereotype.Component

@Component
class EditedMessageUnpackerParser : UnpackerParser<EditedMessageReceiver, Message>() {
    override val updateUnpacker: UpdateUnpacker<Message>
        get() = UpdateUnpacker({ it.editedMessage }, Message::class)

    override val parserAnnotationClass = EditedMessageReceiver::class

}



